silverback/_cli.py

```diff
@@ -241,6 +246,114 @@ def workspaces(platform: PlatformClient):
         )
 
 
+@workspaces.command(name="new", section="Platform Commands (https://silverback.apeworx.io)")
+@click.option(
+    "-n",
+    "--name",
+    "workspace_name",
+    required=True,
+    help="Name for new workspace",
+)
+@click.option(
+    "-s",
+    "--slug",
+    "workspace_slug",
+    required=True,
+    help="Slug for new workspace",
+)
+@platform_client
+def new_workspace(
+    platform: PlatformClient,
+    workspace_name: str,
+    workspace_slug: str,
+):
+    """Create a new workspace"""
+
+    if workspace_name:
+        click.echo(f"name: {workspace_name}")
+        click.echo(f"slug: {workspace_slug or workspace_name.lower().replace(' ', '-')}")
+
+    elif workspace_slug:
+        click.echo(f"slug: {workspace_slug}")
+
+    else:
+        raise click.UsageError("Must provide a name or a slug/name combo")
+
+    platform.create_workspace(
+        workspace_name=workspace_name,
+        workspace_slug=workspace_slug,
+    )
+    click.echo(f"{click.style('SUCCESS', fg='green')}: Created '{workspace_name}'")
+
+
+@workspaces.command(name="delete", section="Platform Commands (https://silverback.apeworx.io)")
+@click.argument("workspace")
+@platform_client
+def delete_workspace(platform: PlatformClient, workspace: str):
+    """Delete an empty Workspace on the Silverback Platform"""
+
+    if not (workspace_client := platform.workspaces.get(workspace)):
+        raise click.BadOptionUsage("workspace", f"Unknown workspace '{workspace}'")
+
+    if list(workspace_client.clusters):
+        raise click.ClickException("Clusters found in Workspace")
+
+    else:
+        platform.remove_workspace(workspace)
+        click.echo(f"{click.style('SUCCESS', fg='green')}: Deleted '{workspace}'")
+
+
+@workspaces.command(name="update", section="Platform Commands (https://silverback.apeworx.io)")
+@click.option(
+    "-n",
+    "--name",
+    "update_name",
+    required=True,
+    help="Update name for workspace",
+)
+@click.option(
+    "-s",
+    "--slug",
+    "update_slug",
+    required=True,
+    help="Update slug for workspace",
+)
+@click.argument("workspace")
+@platform_client
+def update_workspace(
+    platform: PlatformClient,
+    workspace: str,
+    update_name: str,
+    update_slug: str,
```

Comment:
```suggestion
    update_name: str | None,
    update_slug: str | None,
```

Reply:
If the user doesn't have an update name or slug, it seems like it won't keep the previous name/slug. At least that was the case when I tried it using the `/docs/` page, or was that a bug?

Reply:
so if both are empty, it should raise `UsageError` because you haven't tried to update anything

are you saying if one is empty and the other is intended to be changed, they both update?

Reply:
Originally it was if one is empty and the other is intended then they both updated. The new push I made should have fixed that issue though I think.

Reply:
the service itself should allow `None` values and basically make it "no change required", so you shouldn't have to set the values if they are not set and it should work appropiately

my point above is that if both are `None` that is a no-op (won't make any change when you send the PATCH request to the service) so it should raise `click.UsageError` that it's a no-op